src/Microsoft.AspNetCore.Rewrite/RuleAbstaction/ModRewriteRule.cs

```diff
@@ -0,0 +1,165 @@
+            }
+
+            if (Flags.CheckFlag(RuleFlagType.Forbidden) != null)
+            {
+                context.HttpContext.Response.StatusCode = 403;
+                return new RuleResult { Result = RuleTerminiation.ResponseComplete };
+            }
+            else if (Flags.CheckFlag(RuleFlagType.Gone) != null)
+            {
+                context.HttpContext.Response.StatusCode = 410;
+                return new RuleResult { Result = RuleTerminiation.ResponseComplete };
+            }
+            else if (result == "-")
+            {
+                // TODO set url to result.
+            }
+            else if (Flags.CheckFlag(RuleFlagType.QSAppend) != null)
+            {
+                context.HttpContext.Request.QueryString = context.HttpContext.Request.QueryString.Add(new QueryString(result));
+            } 
+            else
+            {
+                context.HttpContext.Request.Path = new PathString(result);
+            }
+            
+            if (Flags.CheckFlag(RuleFlagType.Cookie) != null)
+            {
+                // TODO CreateCookies(context);
+                // context.HttpContext.Response.Cookies.Append()
+            }
+
+            if (Flags.CheckFlag(RuleFlagType.Env) != null)
+            {
+                // TODO CreateEnv(context)
+                // context.HttpContext...
+            }
+
+            if (Flags.CheckFlag(RuleFlagType.Next) != null)
+            {
+
+            }
+
+            string statusCode;
+            if ((statusCode = Flags.CheckFlag(RuleFlagType.Redirect)) != null)
+            {
+                return new RuleResult { Result = RuleTerminiation.ResponseComplete };
+            }
+            else if (Flags.CheckFlag(RuleFlagType.Last) != null || Flags.CheckFlag(RuleFlagType.End) != null)
+            {
+                return new RuleResult { Result = RuleTerminiation.StopRules };
+            }
+            else
+            {
+                return new RuleResult { Result = RuleTerminiation.Continue };
+            }
+        }
+
+        private bool CheckMatchResult(bool? result)
+        {
+            if (result == null)
+            {
+                return false;
+            }
+            return !(result.Value ^ InitialRule.Invert);
+        }
+
+        private bool CheckCondition(ModRewriteRule rule, UrlRewriteContext context, Match results, Match previous)
+        {
+            if (rule.Conditions == null)
+            {
+                return true;
+            }
+
+            // TODO Visitor pattern here
+            foreach (var condition in rule.Conditions)
+            {
+                var concatTestString = condition.TestStringSegments.ApplyPattern(context.HttpContext, results, previous);
+                var match = condition.ConditionExpression.CheckConditionExpression(context, previous, concatTestString);
+
+                if (match && !(Flags.HasFlag(RuleFlagType.Or)))
```

Comment:
I think this condition is wrong. If the condition fails and it's not an or, we want to break out right?